### PR TITLE
WSR v8 

### PR DESCRIPTION
**Pull request by EgoziE**
WSR8 branch.
Jira EPIC - https://jira.wixpress.com/browse/DSM-293

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -32,12 +32,8 @@ autodocs-registry
 !perfer.config.js
 .DS_Store
 /*.scss
-# we also ignore icons folder to avoid comitting icons export
-# icons that appear in root after build
-/new-icons/
-/src/new-icons/system
-/src/new-icons/*.js
 /bundles
+/new-icons
```
Comment (shlomitc) [inline .gitignore]: should be completely removed

File: src/DatePicker/docs/examples.js [hunk 2]
```diff
@@ -1,8 +1,18 @@
 export const simple = `
```
Comment (shlomitc) [inline src/DatePicker/docs/examples.js]: can this be done in `master`?

File: src/Dropdown/docs/examples.js [hunk 3]
```diff
@@ -54,15 +54,15 @@ export const divider = `
 export const prefix = `
   <Dropdown
     placeholder="Select an option"
-    prefix={<Input.Unit>$</Input.Unit>}
+    prefix={<Input.Affix>$</Input.Affix>}
```
Comment (shlomitc) [inline src/Dropdown/docs/examples.js]: probably should be done in master

File: src/ModalSelectorLayout/ModalSelectorLayout.driver.js [hunk 4]
```diff
@@ -1,13 +1,27 @@
...
 import { testkitFactoryCreator } from 'wix-ui-test-utils/vanilla';
 import loaderDriverFactory from '../Loader/Loader.driver';
-import buttonDriverFactory from '../Deprecated/Button/Button.driver';
 import selectorDriverFactory from '../Selector/Selector.driver';
 import searchDriverFactory from '../Search/Search.driver';
 import textDriverFactory from '../Text/Text.driver';
 import { dataHooks } from './ModalSelectorLayout.helpers';
 import checkboxDriverFactory from '../Checkbox/Checkbox.driver';
 
+const buttonDriverFactory = ({ element }) => {
```
Comment (shlomitc) [inline src/ModalSelectorLayout/ModalSelectorLayout.driver.js]: I think this should be a `Button/Button.legacy.driver.js`, next to the unidriver

File: src/ModalSelectorLayout/ModalSelectorLayout.protractor.driver.js [hunk 5]
```diff
@@ -1,10 +1,32 @@
...
+
+import { mergeDrivers } from '../../test/utils/private-drivers';
+import { hasAttribute } from '../../test/utils/protractor-helpers';
 import loaderDriverFactory from '../Loader/Loader.protractor.driver';
-import buttonDriverFactory from '../Deprecated/Button/Button.protractor.driver';
 import textDriverFactory from '../Text/Text.protractor.driver';
 import searchDriverFactory from '../Search/Search.protractor.driver';
 import { dataHooks } from './ModalSelectorLayout.helpers';
 
+const buttonDriverFactory = element => {
```
Comment (shlomitc) [inline src/ModalSelectorLayout/ModalSelectorLayout.protractor.driver.js]: can't we use the `unidriver` here instead?

File: src/Page/docs/ExampleStickyTableWithGap.js [hunk 6]
```diff
@@ -308,13 +307,14 @@ const renderPageHeader = () => {
       <Box>
         <Box>
           <PopoverMenu
-            buttonTheme="icon-greybackground"
-            placement="bottom"
-            size="normal"
-            appendToParent
+            triggerElement={
```
Comment (shlomitc) [inline src/Page/docs/ExampleStickyTableWithGap.js]: this is an amazing example to the migration guide of `PopoverMenu`

File: MIGRATION.md [hunk 7]
```diff
@@ -0,0 +1,1175 @@
...
+Props:
+- removed `error` & `errorMessage` to `status` & `statusMessage`. <br/>
+  For more information and examples go to [error --> status](#error--errormessage----status--statusmessage) section.
+- removed `help` and `helpMessage` use a `<FormField/>` wrapper instead. <br/>
+  For more information and examples go to [help prop](#help-prop) section.
+
+## \<DropdownLayout/>
+
+Props:
+- removed `theme`
```
Comment (shlomitc) [inline MIGRATION.md]: @EgoziE - explain that the past themes are no longer supported by the design system.

File: .storybook/webpack.config.js [hunk 8]
```diff
@@ -36,6 +36,8 @@ module.exports = ({ config }) => {
             issueURL:
               'https://github.com/wix/wix-style-react/issues/new/choose',
             testkitsWarning,
+            importFormat:
```
[the remark this answers is not in the document]
Comment (mykas) [reply .storybook/webpack.config.js]: https://github.com/wix/wix-style-react/pull/5196

File: src/Accordion/docs/index.story.js [hunk 9]
```diff
@@ -93,7 +93,7 @@ export default {
         sections: [
           description(`Component for collapsible content`),
 
-          importExample("import Accordion from 'wix-style-react/Accordion';"),
+          importExample("import { Accordion } from 'wix-style-react';"),
```
[the remark this answers is not in the document]
Comment (mykas) [reply src/Accordion/docs/index.story.js]: https://github.com/wix/wix-style-react/pull/5196

File: src/Accordion/test/Accordion.tsx [hunk 10]
```diff
@@ -1,8 +1,8 @@
 import * as React from 'react';
-import Accordion from '../../src/Accordion';
-import { accordionTestkitFactory } from '../../dist/testkit';
-import { accordionTestkitFactory as accordionEnzymeTestkitFactory } from '../../dist/testkit/enzyme';
-import { accordionTestkitFactory as accordionPuppeteerTestkitFactory } from '../../dist/testkit/puppeteer';
+import Accordion from '..';
```
Comment (shlomitc) [inline src/Accordion/test/Accordion.tsx]: This file structure change should have been done in `master`
@EgoziE

File: src/AddItem/AddItem.js [hunk 11]
```diff
@@ -126,22 +105,34 @@ class AddItem extends Component {
       size,
       disabled,
       showIcon,
-      tooltipContent,
+      tooltipProps = {},
```
Comment (shlomitc) [inline src/AddItem/AddItem.js]: A better approach would have been adding this prop in `master` with some deprecation log

File: src/AddItem/AddItem.st.css [hunk 12]
```diff
@@ -126,6 +126,7 @@
 .root:theme(image) {
   display: block;
   height: 100%;
+  min-height: 81px;
```
Comment (shlomitc) [inline src/AddItem/AddItem.st.css]: @EgoziE - why?

File: src/AddItem/AddItem.st.css [hunk 13]
```diff
@@ -147,10 +148,19 @@
   background-color: value(D70);
 }
 
+/* Tooltip */
+.tooltip {
```
Comment (shlomitc) [inline src/AddItem/AddItem.st.css]: Are you sure tooltip css should be overridden this way? it has its own API regarding positioning
@EgoziE

File: src/FormField/README.TESTKIT.md [hunk 14]
```diff
@@ -11,7 +11,6 @@
 |--------|-----------|----------------|-------------|
 | element | - | element | returns the element |
 | getLabelValue | - | string | get the label text |
-| getTooltipInfoValue | number | string | get the info tooltip text |
```
Comment (shlomitc) [inline src/FormField/README.TESTKIT.md]: I don't see it the migration guide @EgoziE

File: src/GoogleAddressInput/GoogleAddressInput.js [hunk 15]
```diff
@@ -333,8 +332,12 @@ GoogleAddressInput.propTypes = {
   /** Fields indicating which types of Places data to return (see [here](https://developers.google.com/maps/documentation/javascript/places#place_details)**/
   placeDetailsFields: PropTypes.array,
 
-  /** Should display error marker */
-  error: PropTypes.bool,
+  /** Sets UI to indicate a status */
+  status: PropTypes.oneOf(['error', 'warning', 'loading']),
```
Comment (shlomitc) [inline src/GoogleAddressInput/GoogleAddressInput.js]: can't this be done in master?
@EgoziE

File: src/ImageViewer/test/ImageViewer.visual.js [hunk 16]
```diff
@@ -7,50 +7,72 @@ const imageUrl =
 
 const tests = [
   {
-    it: 'with an image',
-    props: {
-      imageUrl,
-    },
+    description: 'standard',
```
Comment (shlomitc) [inline src/ImageViewer/test/ImageViewer.visual.js]: This is a bit dangerous. adding the `description` means you changed the test name, meaning the baseline is no longer relevant @EgoziE

File: src/Input/Input.driver.js [hunk 17]
```diff
@@ -119,6 +104,32 @@ const inputDriverFactory = ({ element }) => {
     isCustomInput: () => {
       return input.getAttribute('data-hook') === 'wsr-custom-input';
     },
+
+    // Status
```
Comment (shlomitc) [inline src/Input/Input.driver.js]: this could have been done in master as well @EgoziE

File: src/Accordion/test/Accordion.tsx (hunk 10 above)
Comment (EgoziE) [reply src/Accordion/test/Accordion.tsx]: Unfortunately we decided to do this when the change was too late..
If it's not critical, I'll leave it like this 🤐

File: src/AddItem/AddItem.st.css (hunk 13 above)
Comment (EgoziE) [reply src/AddItem/AddItem.st.css]: This is a special case, I had no choice 😞 
`<AddItem />` had an inner component `<TooltipHOC />` that changed inner tooltip style in this exact way. I simplified it as much as I could to remove over-complexed code.

File: src/AddItem/AddItem.js (hunk 11 above)
Comment (EgoziE) [reply src/AddItem/AddItem.js]: https://github.com/wix/wix-style-react/pull/5242

File: src/AddItem/AddItem.st.css (hunk 12 above)
Comment (EgoziE) [reply src/AddItem/AddItem.st.css]: same comment as below

File: src/GoogleAddressInput/GoogleAddressInput.js (hunk 15 above)
Comment (EgoziE) [reply src/GoogleAddressInput/GoogleAddressInput.js]: https://github.com/wix/wix-style-react/pull/5243

File: src/Input/Input.driver.js (hunk 17 above)
Comment (EgoziE) [reply src/Input/Input.driver.js]: The change is too deep.
I can try if it's critical

File: src/AddItem/AddItem.st.css (hunk 13 above)
Comment (mykas) [reply src/AddItem/AddItem.st.css]: What kind of special case this is? Tooltip positioning can never be changed. I don't see these styles in the master, why you changed it here?

Comment (mykas) [reply src/AddItem/AddItem.st.css]: This is TooltipHoc in master: https://github.com/wix/wix-style-react/blob/master/src/AddItem/components/TooltipHOC.st.css#L8

File: src/AddItem/AddItem.js [hunk 18]
```diff
@@ -126,22 +105,34 @@ class AddItem extends Component {
...
+          disabled,
+          tooltip: !!tooltipProps.content,
+        })}
+      >
         {showIcon && this._renderIcon()}
         {this._renderText()}
       </div>
     );
-    return (
-      <TooltipHOC
```
Comment (mykas) [inline src/AddItem/AddItem.js]: I think this change is totally unnecessary. The pattern might look different for you but this is just a another pattern

File: src/ModalSelectorLayout/ModalSelectorLayout.protractor.driver.js (hunk 5 above)
Comment (EgoziE) [reply src/ModalSelectorLayout/ModalSelectorLayout.protractor.driver.js]: https://github.com/wix/wix-style-react/pull/5240

File: package.json [hunk 19]
```diff
@@ -1,7 +1,7 @@
 {
   "name": "wix-style-react",
   "description": "wix-style-react",
-  "version": "7.49.1",
+  "version": "8.0.0",
```
[the remark this answers is not in the document]
Comment (EgoziE) [reply package.json]: changelog created

File: src/AddItem/AddItem.st.css (hunk 13 above)
Comment (EgoziE) [reply src/AddItem/AddItem.st.css]: width: 100%;
  height: 100%;
of the tooltip element made its container stretch in height.
This behavior cannot be achieved with current Tooltip

Bottom line, what are you suggesting to do now?

File: src/CircularProgressBar/CircularProgressBar.js [hunk 20]
```diff
@@ -4,9 +4,9 @@ import CircleLoaderCheck from 'wix-ui-icons-common/system/CircleLoaderCheck';
 import CircleLoaderCheckSmall from 'wix-ui-icons-common/system/CircleLoaderCheckSmall';
```
[the remark this answers is not in the document]
Comment (EgoziE) [reply src/CircularProgressBar/CircularProgressBar.js]: done

File: src/Animations/index.js [hunk 21]
```diff
@@ -1 +0,0 @@
-export { default as SlideAnimation } from './SlideAnimation.js';
```
[the remark this answers is not in the document]
Comment (EgoziE) [reply src/Animations/index.js]: done

File: src/ImageViewer/test/ImageViewer.visual.js (hunk 16 above)
Comment (EgoziE) [reply src/ImageViewer/test/ImageViewer.visual.js]: https://github.com/wix/wix-style-react/pull/5248

File: src/AddItem/AddItem.st.css (hunk 13 above)
Comment (mykas) [reply src/AddItem/AddItem.st.css]: That is exactly what we want with Additem it needs to stretch both height and width, whats the problem?

Comment (mykas) [reply src/AddItem/AddItem.st.css]: Those percentage are to make the AddItem work like this:
Screenshot 2020-03-23 at 14 28 48

Comment (mykas) [reply src/AddItem/AddItem.st.css]: Here is another example working in v7
Screenshot 2020-03-23 at 14 30 39

**Comment by TalZa1810** [comment]
yooohoooo!!!

**Pull request merged**